core/consensus/beefy/impl/beefy_impl.cpp

```diff
@@ -393,9 +399,79 @@ namespace kagome::network {
     return outcome::success();
   }
 
+  void BeefyImpl::fetchHeaders() {
+    if (not fetching_headers_ or not beefy_genesis_) {
+      return;
+    }
+
+    while (fetching_headers_) {
+      if (auto& finalized = fetching_headers_->finalizationEstablished; not finalized) {
+        finalized = block_tree_->isFinalized(fetching_headers_->block);
+      }
+      const auto blockNumber = fetching_headers_->block.number;
+      if (blockNumber <= *beefy_genesis_) {
+        fetching_headers_.reset();
+        return;
+      }
+      auto block_hash = block_tree_->getBlockHash(blockNumber);
+      if (not block_hash) {
+        break;
+      }
+      auto& blockHashOptValue = block_hash.value();
+      if (not blockHashOptValue) {
+        break;
+      }
+      auto blockHeader = block_tree_->getBlockHeader(*blockHashOptValue);
+      if (not blockHeader) {
+        break;
+      }
+
+      const auto& blockHeaderValue = blockHeader.value();
+      if (beefyValidatorsDigest(blockHeaderValue)) {
+        beefy_justification_protocol_.get()->fetchJustification(blockNumber);
+      }
+
+      if (const auto parentInfo = blockHeaderValue.parentInfo(); parentInfo) {
+        fetching_headers_->block = *parentInfo;
+      } else {
+        SL_ERROR(log_, "Failed to get parent info for block {}, fetching stopped", blockNumber);
+        fetching_headers_.reset();
+        return;
+      }
```

Comment:
`if (blockNumber <= *beefy_genesis_)` will stop before this point
```suggestion
      fetching_headers_->block = *blockHeaderValue.parentInfo();
```

Reply:
It's just a check for nullopt, why risk?